app/assets/javascripts/discourse/app/components/topic-map/topic-map-summary.gjs

```diff
@@ -1,162 +1,443 @@
+import TopicViewsChart from "discourse/components/topic-map/topic-views-chart";
+import avatar from "discourse/helpers/bound-avatar-template";
 import number from "discourse/helpers/number";
 import slice from "discourse/helpers/slice";
+import { ajax } from "discourse/lib/ajax";
+import { emojiUnescape } from "discourse/lib/text";
+import dIcon from "discourse-common/helpers/d-icon";
 import i18n from "discourse-common/helpers/i18n";
-import { avatarImg } from "discourse-common/lib/avatar-utils";
+import I18n from "discourse-i18n";
+import DMenu from "float-kit/components/d-menu";
+import and from "truth-helpers/helpers/and";
+import lt from "truth-helpers/helpers/lt";
+import not from "truth-helpers/helpers/not";
+
+const TRUNCATED_LINKS_LIMIT = 5;
+const MIN_POST_READ_TIME = 4;
+const MIN_LIKES_COUNT = 5;
+const MIN_POSTS_COUNT = 10;
+const MIN_PARTICIPANTS_COUNT = 5;
+const MIN_USERS_COUNT_FOR_AVATARS = 2;
 
 export default class TopicMapSummary extends Component {
-  get linksCount() {
-    return this.args.topicDetails.links?.length ?? 0;
+  @service site;
+  @service siteSettings;
+  @service mapCache;
+
+  @tracked allLinksShown = false;
+  @tracked top3LikedPosts = [];
+  @tracked views = [];
+  @tracked loading = true;
+
+  get shouldShowParticipants() {
+    return (
+      this.args.topic.posts_count >= MIN_POSTS_COUNT &&
+      this.args.topicDetails.participants?.length >=
+        MIN_USERS_COUNT_FOR_AVATARS &&
+      !this.site.mobileView
+    );
   }
 
-  get createdByUsername() {
-    return this.args.topicDetails.created_by?.username;
+  get readTime() {
+    const calculatedTime = Math.ceil(
+      Math.max(
+        this.args.topic.word_count / this.siteSettings.read_time_word_count,
+        (this.args.topic.posts_count * MIN_POST_READ_TIME) / 60
+      )
+    );
+
+    return calculatedTime > 3 ? calculatedTime : null;
   }
 
-  get lastPosterUsername() {
-    return this.args.topicDetails.last_poster?.username;
+  get topRepliesSummaryEnabled() {
+    return this.args.postStream.summary;
   }
 
-  get toggleMapButton() {
-    return {
-      title: this.args.collapsed
-        ? "topic.expand_details"
-        : "topic.collapse_details",
-      icon: this.args.collapsed ? "chevron-down" : "chevron-up",
-      ariaExpanded: this.args.collapsed ? "false" : "true",
-      ariaControls: "topic-map-expanded__aria-controls",
-      action: this.args.toggleMap,
-    };
+  get topRepliesTitle() {
+    if (this.topRepliesSummaryEnabled) {
+      return;
+    }
+
+    return I18n.t("summary.short_title");
   }
 
-  get shouldShowParticipants() {
+  get topRepliesIcon() {
+    return this.topRepliesSummaryEnabled ? "arrows-alt-v" : "layer-group";
+  }
+
+  get topRepliesLabel() {
+    return this.topRepliesSummaryEnabled
+      ? I18n.t("summary.show_all_label")
+      : I18n.t("summary.short_label");
+  }
+
+  get loneStat() {
+    const hasViews = this.args.topic.views >= 0;
```

Comment:
Should it be more than 1? Since the poster already viewed the post they created (I think)

```suggestion
    const hasViews = this.args.topic.views > 1;
```

Reply:
We still show the view count even if it's 0, although it's not very likely to be the case, as you said